### PR TITLE
Add asciidoc guide for using in Angular-CLI webpack applications

### LINKED ISSUE #21: Add Support for Angular-CLI
- If the user installs the directive using `ng install angular2-polymer` everything in the project should be set to use the elements.
- Additionally we could add some kind of support for scaffolding.
- Everything should be set for vulcanizing when running `ng build`

### PR DESCRIPTION
**Pull request by platosha**
Note for reviewers: The Webpack Version of Angular-CLI is a pre-release. We should probably not publish the guide on vaadin.com/docs until it is an official stable release. Hence the webpack asciidoc guide doesn’t have the docs website header.

Fixes #21

---

This change is [Reviewable](https://reviewable.io/reviews/vaadin/angular2-polymer/65)

### CODE REVIEW

**Comment by tehapo** [comment]
Overall very understandable and a good tutorial, nicely done. :+1:
Added some minor line comments and created #68. :smile: 

In addition I would also add a line break after every sentence instead of having very long lines with multiple sentences. This makes the source file easier to maintain (without horizontal scrolling or having to enable line wrapping). This was also a convention originally suggested by @magi42 

---

Reviewed 2 of 2 files at r1.
Review status: 1 of 2 files reviewed at latest revision, 9 unresolved discussions.

---

_[docs/ng-cli-webpack.adoc, line 26 [r1]](https://reviewable.io:443/reviews/vaadin/angular2-polymer/65#-KPr6NVyZiTdUfGUydwM:-KPr6NVyZiTdUfGUydwN:b-yyeo7g) ([raw file](https://github.com/vaadin/angular2-polymer/blob/6838faa7c5979b0d870ec8c907d8be28e3b7e453/docs/ng-cli-webpack.adoc#L26)):_

> ``` AsciiDoc
> [subs="normal"]
> ----
> [prompt]#$# [command]#npm# install angular-cli -g@webpack
> ```

I think this would be more readable by having the version next to the package name, like the following example.
`npm install -g angular-cli@webpack`

Although I guess the `@webpack` version specifier will be removed when the next version of Angular CLI is released.

---

_[docs/ng-cli-webpack.adoc, line 37 [r1]](https://reviewable.io:443/reviews/vaadin/angular2-polymer/65#-KPr9cu_NodI6vEjpjcx:-KPr9cu_NodI6vEjpjcy:b9y0gyy) ([raw file](https://github.com/vaadin/angular2-polymer/blob/6838faa7c5979b0d870ec8c907d8be28e3b7e453/docs/ng-cli-webpack.adoc#L37)):_

> ``` AsciiDoc
> [prompt]#$# [command]#ng# build
> [prompt]#$# [command]#ng# test
> [prompt]#$# [command]#ng# serve
> ```

Should we mention that after this the app is running at `http://localhost:4200`? It is mentioned in the output, but might be lost in the verbose logging.

---

_[docs/ng-cli-webpack.adoc, line 52 [r1]](https://reviewable.io:443/reviews/vaadin/angular2-polymer/65#-KPr7fQbrNZIL011Api6:-KPr7fQbrNZIL011Api7:bhp78u8) ([raw file](https://github.com/vaadin/angular2-polymer/blob/6838faa7c5979b0d870ec8c907d8be28e3b7e453/docs/ng-cli-webpack.adoc#L52)):_

> ``` AsciiDoc
> ----
> 
> By default, Bower installs dependencies in the [filename]#bower_components# folder. But Angular-CLI expects static stuff to be in the [filename]#public# directory. Thus, create the [filename]#.bowerrc# file in the root directory, with the following content:
> ```

1) I think this should be "installs dependencies _to_ the".

2) Maybe the `.bowerrc` file should be elaborated a bit more. Something along the lines of:
"Create a .bowerrc file to instruct Bower to install dependencies inside the directory expected by Angular."

---

_[docs/ng-cli-webpack.adoc, line 82 [r1]](https://reviewable.io:443/reviews/vaadin/angular2-polymer/65#-KPrERc-duO5xKjztGFZ:-KPrERc0L74hwEI7Iljg:b-pkblf5) ([raw file](https://github.com/vaadin/angular2-polymer/blob/6838faa7c5979b0d870ec8c907d8be28e3b7e453/docs/ng-cli-webpack.adoc#L82)):_

> ``` AsciiDoc
> ====
> 
> Next we will modify the application index file to load the web components polyfill and import the Polymer elements. Open the [filename]#src/index.html# file and append the following lines to the [elementname]#head# section as follows:
> ```

1) Not sure if the "index file" needs to be mentioned here. IMO the text would flow better with just "we will modify the application to load the web components...".

The `src/index.html` is mentioned in the next sentence anyways.

2) Repeated "following/follows". Maybe just say "append the following lines to the head section." (with removing the "as follows").

---

_[docs/ng-cli-webpack.adoc, line 98 [r1]](https://reviewable.io:443/reviews/vaadin/angular2-polymer/65#-KPrFPZzJEV8hhvGk874:-KPrFPZzJEV8hhvGk875:b-1u81bf) ([raw file](https://github.com/vaadin/angular2-polymer/blob/6838faa7c5979b0d870ec8c907d8be28e3b7e453/docs/ng-cli-webpack.adoc#L98)):_

> ``` AsciiDoc
> ----
> 
> In Angular-CLI Webpack projects, the main application file is automatically bundled and appended to the end of the [elementname]#body# section in the application’s [filename]#index.html# file. It means that the Angular application is imported and bootstrapped synchronousely. Meanwhile, the Polymer elemnts are loaded from the HTML Imports, which are processed asynchronousely in browsers that do not have a native support.
> ```

Couple of typos.
synchronousely → synchronously
asynchronousely → asynchronously

---

_[docs/ng-cli-webpack.adoc, line 100 [r1]](https://reviewable.io:443/reviews/vaadin/angular2-polymer/65#-KPrFmBunB2Gm8IA0mxU:-KPrFmBunB2Gm8IA0mxV:b4304sn) ([raw file](https://github.com/vaadin/angular2-polymer/blob/6838faa7c5979b0d870ec8c907d8be28e3b7e453/docs/ng-cli-webpack.adoc#L100)):_

> ``` AsciiDoc
> In Angular-CLI Webpack projects, the main application file is automatically bundled and appended to the end of the [elementname]#body# section in the application’s [filename]#index.html# file. It means that the Angular application is imported and bootstrapped synchronousely. Meanwhile, the Polymer elemnts are loaded from the HTML Imports, which are processed asynchronousely in browsers that do not have a native support.
> 
> In order to use Polymer elements from javascript with cross-browser support, we have to wait for the Polymer elements to be loaded and registered before running the application code. Therefore, we have to postpone the Angular application import after the [eventname]#WebComponentsReady# event is dispatched. Add the loader script file [filename]#src/main-polymer.ts# with the following contents:
> ```

javascript → Javascript

The last sentence sounds a bit weird. I would reword this a bit. Maybe something like:
"Create a new file src/main-polymer.ts with the following loader script."

---

_[docs/ng-cli-webpack.adoc, line 143 [r1]](https://reviewable.io:443/reviews/vaadin/angular2-polymer/65#-KPrI-ejDDESJGIZKI6P:-KPrI-ejDDESJGIZKI6Q:b-occfx3) ([raw file](https://github.com/vaadin/angular2-polymer/blob/6838faa7c5979b0d870ec8c907d8be28e3b7e453/docs/ng-cli-webpack.adoc#L143)):_

> ``` AsciiDoc
> == Using Polymer Elements
> 
> Now that everything is set, we can add any polymer elements to our application using their element names in templates, and the [classname]#PolymerElement# directive in code.
> ```

polymer → Polymer

---

_[docs/ng-cli-webpack.adoc, line 162 [r1]](https://reviewable.io:443/reviews/vaadin/angular2-polymer/65#-KPrJfIcnRKPEhOJvcMW:-KPrJfIcnRKPEhOJvcMX:b-4b5q7g) ([raw file](https://github.com/vaadin/angular2-polymer/blob/6838faa7c5979b0d870ec8c907d8be28e3b7e453/docs/ng-cli-webpack.adoc#L162)):_

> ``` AsciiDoc
> 
> @Component({
>   moduleId: module.id,
> ```

I had to remove this line `moduleId: module.id,` to make the app work.

Got this error otherwise:
https://github.com/angular/angular/issues/10626

---

_[docs/ng-cli-webpack.adoc, line 208 [r1]](https://reviewable.io:443/reviews/vaadin/angular2-polymer/65#-KPrK1Vs_PYSkcFQXxnR:-KPrK1Vs_PYSkcFQXxnS:b5z5l5q) ([raw file](https://github.com/vaadin/angular2-polymer/blob/6838faa7c5979b0d870ec8c907d8be28e3b7e453/docs/ng-cli-webpack.adoc#L208)):_

> ``` AsciiDoc
> ----
> 
> Finally, you can use Polymer custom CSS properties and custom CSS mixins either: in the [filename]#app.component.ts# file for the scoped styles, or in the [filename]#index.html# file for the global ones.
> ```

The word "either" looks a bit weird here. Maybe just remove it?

---

_Comments from [Reviewable](https://reviewable.io:443/reviews/vaadin/angular2-polymer/65#-:-KPrGp7Za355lgsaptIK:ba8gkvm)_

**Comment by tehapo** [comment]
Review status: 1 of 2 files reviewed at latest revision, 9 unresolved discussions.

---

_[docs/ng-cli-webpack.adoc, line 100 [r1]](https://reviewable.io:443/reviews/vaadin/angular2-polymer/65#-KPrFmBunB2Gm8IA0mxU:-KPra_SGaAooq6SxjG6O:b-rjslsq) ([raw file](https://github.com/vaadin/angular2-polymer/blob/6838faa7c5979b0d870ec8c907d8be28e3b7e453/docs/ng-cli-webpack.adoc#L100)):_
<details><summary>

<i>Previously, tehapo (Teemu Pöntelin) wrote…</i></summary>



> javascript → Javascript
> 
> The last sentence sounds a bit weird. I would reword this a bit. Maybe something like:
> "Create a new file src/main-polymer.ts with the following loader script."
> </details>
> Actually "javascript" → "JavaScript" :smile: 

---

_Comments from [Reviewable](https://reviewable.io:443/reviews/vaadin/angular2-polymer/65)_

**Comment by platosha** [comment]
Review status: 1 of 2 files reviewed at latest revision, 9 unresolved discussions.

---

_[docs/ng-cli-webpack.adoc, line 26 [r1]](https://reviewable.io:443/reviews/vaadin/angular2-polymer/65#-KPr6NVyZiTdUfGUydwM:-KPrdnvhRIt9OBhz_YAG:b-896fix) ([raw file](https://github.com/vaadin/angular2-polymer/blob/6838faa7c5979b0d870ec8c907d8be28e3b7e453/docs/ng-cli-webpack.adoc#L26)):_
<details><summary>

<i>Previously, tehapo (Teemu Pöntelin) wrote…</i></summary>



> I think this would be more readable by having the version next to the package name, like the following example.
> `npm install -g angular-cli@webpack`
> 
> Although I guess the `@webpack` version specifier will be removed when the next version of Angular CLI is released.
> </details>
> Done.

---

_[docs/ng-cli-webpack.adoc, line 37 [r1]](https://reviewable.io:443/reviews/vaadin/angular2-polymer/65#-KPr9cu_NodI6vEjpjcx:-KPrdoguahpGHK6OleLf:b-896fix) ([raw file](https://github.com/vaadin/angular2-polymer/blob/6838faa7c5979b0d870ec8c907d8be28e3b7e453/docs/ng-cli-webpack.adoc#L37)):_
<details><summary>

<i>Previously, tehapo (Teemu Pöntelin) wrote…</i></summary>



> Should we mention that after this the app is running at `http://localhost:4200`? It is mentioned in the output, but might be lost in the verbose logging.
> </details>
> Done.

---

_[docs/ng-cli-webpack.adoc, line 52 [r1]](https://reviewable.io:443/reviews/vaadin/angular2-polymer/65#-KPr7fQbrNZIL011Api6:-KPrdpkqI739_IDwJDAy:byawbvv) ([raw file](https://github.com/vaadin/angular2-polymer/blob/6838faa7c5979b0d870ec8c907d8be28e3b7e453/docs/ng-cli-webpack.adoc#L52)):_
<details><summary>

<i>Previously, tehapo (Teemu Pöntelin) wrote…</i></summary>



> 1) I think this should be "installs dependencies _to_ the".
> 
> 2) Maybe the `.bowerrc` file should be elaborated a bit more. Something along the lines of:
> "Create a .bowerrc file to instruct Bower to install dependencies inside the directory expected by Angular."
> </details>
> 1) Done.

2) No action. If I extend the last sentence, it will become too complex. And if I make a separate sentence, it duplicates the meaning of the other sentences in this paragraph.

---

_[docs/ng-cli-webpack.adoc, line 82 [r1]](https://reviewable.io:443/reviews/vaadin/angular2-polymer/65#-KPrERc-duO5xKjztGFZ:-KPreWB9zAFBAqFVQS-z:b-z1rbfn) ([raw file](https://github.com/vaadin/angular2-polymer/blob/6838faa7c5979b0d870ec8c907d8be28e3b7e453/docs/ng-cli-webpack.adoc#L82)):_
<details><summary>

<i>Previously, tehapo (Teemu Pöntelin) wrote…</i></summary>



> 1) Not sure if the "index file" needs to be mentioned here. IMO the text would flow better with just "we will modify the application to load the web components...".
> 
> The `src/index.html` is mentioned in the next sentence anyways.
> 
> 2) Repeated "following/follows". Maybe just say "append the following lines to the head section." (with removing the "as follows").
> </details>
> 1) Reworded as "index HTML file". Just "modify the application" feels too unspecific. Repeating is not an issue here IMO.

2) Done.

---

_[docs/ng-cli-webpack.adoc, line 98 [r1]](https://reviewable.io:443/reviews/vaadin/angular2-polymer/65#-KPrFPZzJEV8hhvGk874:-KPrf6AHHZqptRI8hj1b:b-896fix) ([raw file](https://github.com/vaadin/angular2-polymer/blob/6838faa7c5979b0d870ec8c907d8be28e3b7e453/docs/ng-cli-webpack.adoc#L98)):_
<details><summary>

<i>Previously, tehapo (Teemu Pöntelin) wrote…</i></summary>



> Couple of typos.
> synchronousely → synchronously
> asynchronousely → asynchronously
> </details>
> Done.

---

_[docs/ng-cli-webpack.adoc, line 100 [r1]](https://reviewable.io:443/reviews/vaadin/angular2-polymer/65#-KPrFmBunB2Gm8IA0mxU:-KPrf9o0Z2Iqs10Tt3JM:b-896fix) ([raw file](https://github.com/vaadin/angular2-polymer/blob/6838faa7c5979b0d870ec8c907d8be28e3b7e453/docs/ng-cli-webpack.adoc#L100)):_
<details><summary>

<i>Previously, tehapo (Teemu Pöntelin) wrote…</i></summary>



> Actually "javascript" → "JavaScript" :smile: 
> </details>
> Done.

---

_[docs/ng-cli-webpack.adoc, line 143 [r1]](https://reviewable.io:443/reviews/vaadin/angular2-polymer/65#-KPrI-ejDDESJGIZKI6P:-KPrfAwBe2-AGyOCgymG:b-896fix) ([raw file](https://github.com/vaadin/angular2-polymer/blob/6838faa7c5979b0d870ec8c907d8be28e3b7e453/docs/ng-cli-webpack.adoc#L143)):_
<details><summary>

<i>Previously, tehapo (Teemu Pöntelin) wrote…</i></summary>



> polymer → Polymer
> </details>
> Done.

---

_[docs/ng-cli-webpack.adoc, line 162 [r1]](https://reviewable.io:443/reviews/vaadin/angular2-polymer/65#-KPrJfIcnRKPEhOJvcMW:-KPrfBSzcaFy9ainZH6p:b-896fix) ([raw file](https://github.com/vaadin/angular2-polymer/blob/6838faa7c5979b0d870ec8c907d8be28e3b7e453/docs/ng-cli-webpack.adoc#L162)):_
<details><summary>

<i>Previously, tehapo (Teemu Pöntelin) wrote…</i></summary>



> I had to remove this line `moduleId: module.id,` to make the app work.
> 
> Got this error otherwise:
> https://github.com/angular/angular/issues/10626
> </details>
> Done.

---

_[docs/ng-cli-webpack.adoc, line 208 [r1]](https://reviewable.io:443/reviews/vaadin/angular2-polymer/65#-KPrK1Vs_PYSkcFQXxnR:-KPrfF87HFfqRfI-KMkv:b-896fix) ([raw file](https://github.com/vaadin/angular2-polymer/blob/6838faa7c5979b0d870ec8c907d8be28e3b7e453/docs/ng-cli-webpack.adoc#L208)):_
<details><summary>

<i>Previously, tehapo (Teemu Pöntelin) wrote…</i></summary>



> The word "either" looks a bit weird here. Maybe just remove it?
> </details>
> Done.

---

_Comments from [Reviewable](https://reviewable.io:443/reviews/vaadin/angular2-polymer/65)_

**Comment by tehapo** [comment]
:lgtm:

---

Reviewed 1 of 1 files at r2.
Review status: all files reviewed at latest revision, 3 unresolved discussions.

---

_[docs/ng-cli-webpack.adoc, line 37 [r1]](https://reviewable.io:443/reviews/vaadin/angular2-polymer/65#-KPr9cu_NodI6vEjpjcx:-KPuxrDFm8gHaXupPcUL:bbcplm7) ([raw file](https://github.com/vaadin/angular2-polymer/blob/6838faa7c5979b0d870ec8c907d8be28e3b7e453/docs/ng-cli-webpack.adoc#L37)):_
<details><summary>

<i>Previously, platosha (Anton Platonov) wrote…</i></summary>



> Done.
> </details>
> Good idea to also add Ctrl+C there. :+1: 

---

_[docs/ng-cli-webpack.adoc, line 52 [r1]](https://reviewable.io:443/reviews/vaadin/angular2-polymer/65#-KPr7fQbrNZIL011Api6:-KPuxwb9WM_KJagWaYrT:b-r9nx4s) ([raw file](https://github.com/vaadin/angular2-polymer/blob/6838faa7c5979b0d870ec8c907d8be28e3b7e453/docs/ng-cli-webpack.adoc#L52)):_
<details><summary>

<i>Previously, platosha (Anton Platonov) wrote…</i></summary>



> 1) Done.
> 
> 2) No action. If I extend the last sentence, it will become too complex. And if I make a separate sentence, it duplicates the meaning of the other sentences in this paragraph.
> </details>
> 2) Fair enough, I'm ok with that.

---

_[docs/ng-cli-webpack.adoc, line 82 [r1]](https://reviewable.io:443/reviews/vaadin/angular2-polymer/65#-KPrERc-duO5xKjztGFZ:-KPuy2tZr79JIiT5u-NP:b-ro1ova) ([raw file](https://github.com/vaadin/angular2-polymer/blob/6838faa7c5979b0d870ec8c907d8be28e3b7e453/docs/ng-cli-webpack.adoc#L82)):_
<details><summary>

<i>Previously, platosha (Anton Platonov) wrote…</i></summary>



> 1) Reworded as "index HTML file". Just "modify the application" feels too unspecific. Repeating is not an issue here IMO.
> 
> 2) Done.
> </details>
> 1) Ok, that's sounds also good.

---

_Comments from [Reviewable](https://reviewable.io:443/reviews/vaadin/angular2-polymer/65#-:-KPuydw7K1zwQZbeOcGH:bnfp4nl)_

**Comment by tehapo** [comment]
Merging this documentation now. As mentioned in the PR description, this will not appear in the official documentation until we uncomment links and add a proper header metadata. This should be done after Angular CLI releases the Webpack version.

---

Review status: all files reviewed at latest revision, 3 unresolved discussions.

---

_Comments from [Reviewable](https://reviewable.io:443/reviews/vaadin/angular2-polymer/65#-:-KPv1axLVOMRJN0l9B4e:b-i3lu8y)_
